### DISCUSSION TITLE
Section Component Problem

### DISCUSSION THREAD

**Discussion by seoulaja** [Budibase/budibase#11373] [Help] [answered]
The section component is a great one. It helps me to organize other components on the screen. We can only add a Section Component if we are at the highest level of the component structure. But we can make a workaround if we want a section available on the lower level. The steps are: create a section on the highest level, once it's created then we can move the section to whatever level we want as below 

![Screenshot 2023-07-26 at 22 25 09](https://github.com/Budibase/budibase/assets/15101654/69b29caf-a458-4445-9c68-500d53b2019a)

It works perfectly fine.

The problem is when I exported the application and then restore it to another application, it generates an annoying error. It says: "You can't place a Section here"
<img width="349" alt="Screenshot 2023-07-26 at 22 26 54" src="https://github.com/Budibase/budibase/assets/15101654/597d523e-c350-488c-9fac-80080989d583">

I have another application that uses a lot of sections in the lower level, I tried to export the application and restore it in another application, and I have the same error. The frustrating part is I cannot edit or delete the section, it keeps showing me the error: You can't place a section here

Any thoughts?

**Comment by andz-bb** [comment, marked as answer]
hey @seoulaja, sorry to see you are having this issue.

Since the last update we are now enforcing the proper component structure for screens. The section component wasn't designed to be a child of a lot of other components so we have a validation rule that prevents that from being done ([see this line of the code for example](https://github.com/Budibase/budibase/blob/4465c73648ad00fa058efc6aa42dc298aab386f3/packages/client/manifest.json#L240) where the property that prevents this is set)

For context, that ticket to enforce the validation in the latest update can be found here: #10229 

To prevent this change from breaking existing apps like yours that wouldn't pass the new validation rules, it was set to only apply to apps that were created after the update. Unfortunately in your situation, that also included old apps that have been imported after the update. When you imported the app it was being treated as a newly created app so the validation was being applied.

After some discussion with the engineers yesterday, a fix has been written to ensure that the new validation rules don't apply to legacy app exports that have been imported in the new version. The pull request for that fix is here: #11384 

That fix should be merged soon and will be included in the next build of Budibase. Once it is released it should allow you to work with your older apps as normal! Hope this helps

**Comment by seoulaja** [comment]
Hi @andz-bb 

Thank you for your response and the incoming solution for the older apps, really appreciate it!

One of the reasons I love using sections is because it makes the layout easier and it still looks good in mobile view. It would be great if the same behavior can be implemented in the container.

Thank you!
